components/event-bus/internal/knative/subscription/controller/eventactivation/reconcile.go

```diff
@@ -3,114 +3,131 @@ package eventactivation
-	requeue, reconcileErr := r.reconcile(ctx, ea)
+	reconcileErr := r.reconcile(ctx, ea)
 	if reconcileErr != nil {
-		log.Error(reconcileErr, "Reconciling EventActivation")
-		r.recorder.Eventf(ea, corev1.EventTypeWarning, "EventactivationReconcileFailed", "Eventactivation reconciliation failed: %v", reconcileErr)
+		r.Recorder.Eventf(ea, corev1.EventTypeWarning, eventactivationreconciledfailed,
+			"Eventactivation reconciliation failed: %v", reconcileErr)
 	}
 
-	if updateStatusErr := util.UpdateEventActivation(ctx, r.client, ea); updateStatusErr != nil {
-		log.Error(updateStatusErr, "Updating EventActivation status")
-		r.recorder.Eventf(ea, corev1.EventTypeWarning, "EventactivationReconcileFailed", "Updating EventActivation status failed: %v", updateStatusErr)
-		return reconcile.Result{}, updateStatusErr
+	if err := util.UpdateEventActivation(r.applicationconnectorClient, ea); err != nil {
+		r.Recorder.Eventf(ea, corev1.EventTypeWarning, eventactivationreconciledfailed, "Updating EventActivation status failed: %v", err)
+		return err
 	}
 
-	if !requeue && reconcileErr == nil {
-		log.Info("EventActivation reconciled")
-		r.recorder.Eventf(ea, corev1.EventTypeNormal, "EventactivationReconciled", "EventActivation reconciled, name: %q; namespace: %q", ea.Name, ea.Namespace)
+	if reconcileErr == nil {
+		r.Recorder.Eventf(ea, corev1.EventTypeNormal, eventactivationreconciled, "EventActivation reconciled, name: %q; namespace: %q", ea.Name, ea.Namespace)
 	}
-	return reconcile.Result{
-		Requeue: requeue,
-	}, reconcileErr
+
+	return reconcileErr
 }
 
-func (r *reconciler) reconcile(ctx context.Context, ea *eventingv1alpha1.EventActivation) (bool, error) {
+// eventActivationByKey retrieves a EventActivation object from a lister by ns/name key.
+func eventActivationByKey(key string, l applicationconnectorlistersv1alpha1.EventActivationLister) (*applicationconnectorv1alpha1.EventActivation, error) {
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return nil, controller.NewPermanentError(pkgerrors.Wrap(err, "invalid object key"))
+	}
+
+	src, err := l.EventActivations(ns).Get(name)
+	switch {
+	case apierrors.IsNotFound(err):
+		return nil, controller.NewPermanentError(pkgerrors.Wrap(err, "object no longer exists"))
+	case err != nil:
+		return nil, err
+	}
 
+	return src, nil
+}
+
+func (r *Reconciler) reconcile(ctx context.Context, ea *applicationconnectorv1alpha1.EventActivation) error {
+	log := logging.FromContext(ctx)
 	// delete or add finalizers
 	if !ea.DeletionTimestamp.IsZero() {
 		// deactivate all Kyma subscriptions related to this ea
-		subs, _ := util.GetSubscriptionsForEventActivation(ctx, r.client, ea)
-		util.DeactivateSubscriptions(ctx, r.client, subs, log, r.time)
-
+		subs, err := util.GetSubscriptionsForEventActivation(r.kymaEventingClient, ea)
+		if err != nil {
+			return err
+		}
+		error := util.DeactivateSubscriptions(r.kymaEventingClient, subs, log, r.time)
```

Comment:
The name "error" clashes with the `error` interface, quite confusing for the reader. Why not simply assigning to `err =`, which is already declared above?